python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py

```diff
@@ -5,20 +5,21 @@
 import dagster._check as check
 from dagster._record import record
 from jinja2 import Template
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
 from pydantic.fields import FieldInfo
 
 T = TypeVar("T")
 
 REF_BASE = "#/$defs/"
 REF_TEMPLATE = f"{REF_BASE}{{model}}"
 
-CONTEXT_KEY = "required_rendering_context"
+CONTEXT_KEY = "required_rendering_scope"
 
 
-def add_required_rendering_context(field: Any, context: AbstractSet[str]) -> Any:
+def RenderingScope(field: Optional[FieldInfo] = None, *, required_scope: AbstractSet[str]) -> Any:
```

Comment:
under what circumstances will `field` be None? Seems simpler/clearer to make it required.

Reply:
It's a convenience thing, basically if you have:

```python
class Thing(BaseModel):
    thing: Foo
```
and you want to make it have some required scope, it's nicer to do:

```python
class Thing(BaseModel):
    thing: Foo = RenderingScope(required_scope={"abc"})
```

than

```python
class Thing(BaseModel):
    thing: Foo = RenderingScope(Field(), required_scope={"abc"})
```

Essentially, this lets you ignore the Field() bit in cases where you didn't need to use it before

Reply:
roger that makes sense thanks

Would be useful to have these examples in the docblock